File: 169. Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        temp = int(len(nums)/2)
        nums = sorted(nums)
        if len(nums) == 1:
            return nums[0]
        for x in range(len(nums)-1):
            if nums[x] == nums[temp]:
                return nums[x]
        

File: go_solution_set1/169. Majority Element.go
import (
    "math"
)
func majorityElement(nums []int) int {
    counter := make(map[int]int)

    for _, val := range nums{
        _,present := counter[val]
        if present{
            counter[val]+=1
        } else {
            counter[val] = 1
        }
    }

    tmp := int(math.Floor(float64(len(nums))/2))
    
    for key, val := range counter{
        if val > tmp {
            return key
        }
    }

    return 0
}
